graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -104,7 +133,12 @@ object PageRank extends Logging {
       // edge partitions.
       prevRankGraph = rankGraph
       rankGraph = rankGraph.joinVertices(rankUpdates) {
+        if (personalized) {
```

Comment:
It would be better to move the conditional outside the closure (i.e., `if (personalized) { rankGraph.joinVertices ... } else { ...} `) to avoid a branch at each vertex.